From 078d552ed404599c968d13a560178fa874d748e4 Mon Sep 17 00:00:00 2001
From: Rudi <r.ihle@s-t.de>
Date: Wed, 30 Dec 2015 11:16:10 +0100
Subject: [PATCH] video: mxc_edid: Unswap margins for interlaced modes

The left/right and upper/lower margins for interlaced modes are swapped.

Signed-off-by: Rudi <r.ihle@s-t.de>
---
 drivers/video/mxc/mxc_edid.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/video/mxc/mxc_edid.c b/drivers/video/mxc/mxc_edid.c
index 3baac3c..e11fcd9 100644
--- a/drivers/video/mxc/mxc_edid.c
+++ b/drivers/video/mxc/mxc_edid.c
@@ -67,18 +67,18 @@ const struct fb_videomode mxc_cea_mode[64] = {
 	},
 	/* #5: 1920x1080i@59.94/60Hz 16:9 */
 	[5] = {
-		NULL, 60, 1920, 1080, 13468, 88, 148, 4, 31, 44, 10,
+		NULL, 60, 1920, 1080, 13468, 148, 88, 31, 4, 44, 10,
 		FB_SYNC_HOR_HIGH_ACT | FB_SYNC_VERT_HIGH_ACT,
 		FB_VMODE_INTERLACED | FB_VMODE_ASPECT_16_9, 0,
 	},
 	/* #6: 720(1440)x480iH@59.94/60Hz 4:3 */
 	[6] = {
-		NULL, 60, 720, 480, 37000, 38, 114, 8, 31, 124, 6, 0,
+		NULL, 60, 720, 480, 37000, 114, 38, 31, 8, 124, 6, 0,
 		FB_VMODE_INTERLACED | FB_VMODE_ASPECT_4_3 | FB_VMODE_DBLCLK, 0,
 	},
 	/* #7: 720(1440)x480iH@59.94/60Hz 16:9 */
 	[7] = {
-		NULL, 60, 720, 480, 37000, 38, 114, 8, 31, 124, 6, 0,
+		NULL, 60, 720, 480, 37000, 114, 38, 31, 8, 124, 6, 0,
 		FB_VMODE_INTERLACED | FB_VMODE_ASPECT_16_9 | FB_VMODE_DBLCLK, 0,
 	},
 	/* #8: 720(1440)x240pH@59.94/60Hz 4:3 */
@@ -125,18 +125,18 @@ const struct fb_videomode mxc_cea_mode[64] = {
 	},
 	/* #20: 1920x1080i@50Hz */
 	[20] = {
-		NULL, 50, 1920, 1080, 13468, 528, 148, 4, 31, 44, 10,
+		NULL, 50, 1920, 1080, 13468, 148, 528, 31, 4, 44, 10,
 		FB_SYNC_HOR_HIGH_ACT | FB_SYNC_VERT_HIGH_ACT,
 		FB_VMODE_INTERLACED | FB_VMODE_ASPECT_16_9, 0,
 	},
 	/* #21: 720(1440)x576i@50Hz */
 	[21] = {
-		NULL, 50, 720, 576, 37037, 24, 138, 4, 39, 126, 6, 0,
+		NULL, 50, 720, 576, 37037, 138, 24, 39, 4, 126, 6, 0,
 		FB_VMODE_INTERLACED | FB_VMODE_ASPECT_4_3 | FB_VMODE_DBLCLK, 0,
 	},
 	/* #22: 720(1440)x576i@50Hz */
 	[22] = {
-		NULL, 50, 720, 576, 37037, 24, 138, 4, 39, 126, 6, 0,
+		NULL, 50, 720, 576, 37037, 138, 24, 39, 4, 126, 6, 0,
 		FB_VMODE_INTERLACED | FB_VMODE_ASPECT_16_9 | FB_VMODE_DBLCLK, 0,
 	},
 	/* #23: 720(1440)x288pH@50Hz 4:3 */
-- 
1.9.1

